internal/db/repos/instance.go

```diff
@@ -69,13 +69,16 @@ func (r *InstanceRepository) UpdateByID(ctx context.Context, id uint, instance *
 }
 
 // UpdateByName updates an instance by its name. Only non-zero fields in the instance parameter will be updated.
-// See UpdateByID method for details on how GORM handles field updates.
 func (r *InstanceRepository) UpdateByName(ctx context.Context, name string, instance *models.Instance) error {
 	return r.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
-		if err := tx.Model(&models.Instance{}).
-			Where(&models.Instance{Name: name}).
-			Updates(instance).Error; err != nil {
-			return fmt.Errorf("failed to update instance by name: %w", err)
+		result := tx.Model(&models.Instance{}).
+			Where("name = ? AND deleted_at IS NULL", name).
+			Updates(instance)
+		if result.Error != nil {
+			return fmt.Errorf("failed to update instance by name: %w", result.Error)
+		}
+		if result.RowsAffected == 0 {
+			return fmt.Errorf("no instance found with name: %s", name)
```

Comment:
[Nit] Not sure if that is required. I am not 100% sure but I think if user does not change anything and clicks on the save button, they will receive this error because nothing is changed and RowsAffected is zero. need to do some research abt it.